### ISSUE TITLE
`Toolbar`

### ISSUE THREAD

**Issue by GroovinChip** [macosui/macos_ui#80]
<img width="309" alt="Screen Shot 2021-05-13 at 12 01 46 PM" src="https://user-images.githubusercontent.com/4250470/118152697-ff2c3480-b3e2-11eb-8dac-598d38727c5a.png">

<img width="284" alt="Screen Shot 2021-05-13 at 9 29 32 PM" src="https://user-images.githubusercontent.com/4250470/118207280-53aad080-b432-11eb-8ed7-39d9320527f3.png">

* A `TitleBar` should take a `List<Toolbar> toolbars`. A `Toolbar` should therefore be defined as a Widget that can take multiple children and lay them our horizontally according to the developer's preference.

**Comment by whiplashoo** [comment]
OK, so I did some research on how the `Toolbar` widget should be implemented to match Apple design:

A `ToolBar` should be a different widget from the `TitleBar`. A `ToolBar` though, can also be integrated with the `TitleBar` (i.e. it can include the app's title info, traditionally in the beginning). [More info](https://developer.apple.com/design/human-interface-guidelines/macos/windows-and-views/window-anatomy/) and examples:

Keynote app (title bar + tool bar):
 
<img width="1227" alt="Screenshot 2022-04-23 at 12 09 10 AM" src="https://user-images.githubusercontent.com/9117427/164793855-35a0621a-21dc-417e-95bd-0da5187f7687.png">

Sketch app (integrated tool bar):

<img width="1196" alt="Screenshot 2022-04-23 at 12 09 02 AM" src="https://user-images.githubusercontent.com/9117427/164793861-98f0e68e-5bb0-4049-a3e2-ed312c5b4f38.png">

**So, a user should be able to choose and use a preferred window layout:** 

- with a title bar (and optionally, a toolbar below it), 
- with just an integrated toolbar
- without either a title bar or a toolbar (e.g. the Facetime app). 

The issue is that including a title bar affects the placement of the traffic lights buttons (close, minimize, zoom), as shown in above screenshots. So, this should be decided beforehand, as it's necessary to set the `toolbarStyle` setting with Swift code in `MainFlutterWindow.swift`:

```
if #available(macOS 11.0, *) {
  // Use .expanded if the app will have a title bar, else use .unified
  self.toolbarStyle = .expanded
} 
```

So, my plan for implementing the `ToolBar` widget is to:
- Simplify the existing `TitleBar` widget to only hold one widget (most probably, a `Text` widget with the app's name).
- Move the `TitleBar` widget to `MacosWindow` (instead of the `MacosScaffold`), as it should be used app-wide.
- Add a `ToolBar` widget to the Scaffold, similar to the previous `TitleBar` widget, that will hold all the toolbar items, a navigational back button, as well as an optional title slot (like in the Sketch screenshot).
- Communicate how to set the proper toolbar style via Swift code in the docs.
- Implement relevant widgets for the toolbar items.

Of course, all these changes will lead to some breaking changes for users that have been using the title bar until now (including me 😄). 

A first iteration of this:
<img width="886" alt="Screenshot 2022-04-23 at 12 27 05 AM" src="https://user-images.githubusercontent.com/9117427/164797685-2b4e2a4b-7657-49eb-af16-d1af9f271ca9.png">

Any thoughts or feedback would be welcome! @GroovinChip

**Comment by GroovinChip** [comment]
@whiplashoo this all looks great! One comment I have is that we should figure out a way to allow users to choose if the dividers between the titlebar/toolbar/content area should be visible or not, since some apps (like the ones in your screenshots) do not show them.
